### PR TITLE
[FLINK-32850][flink-runtime][JUnit5 Migration] The io.network.api package of flink-runtime module

### PR DESCRIPTION
**Pull request by Jiabao-Sun**
## What is the purpose of the change
[FLINK-32850][flink-runtime][JUnit5 Migration] The io.network.api package of flink-runtime module

## Brief change log
[FLINK-32850][flink-runtime][JUnit5 Migration] The io.network.api package of flink-runtime module

## Verifying this change
This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by leonardBang** [comment]
@dawidwys Would you like to help review this PR?

File: flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/CheckpointBarrierTest.java [hunk 1]
```diff
@@ -22,37 +22,29 @@
...
+    void testSerialization() {
         long id = Integer.MAX_VALUE + 123123L;
         long timestamp = Integer.MAX_VALUE + 1228L;
 
         CheckpointOptions options = CheckpointOptions.forCheckpointWithDefaultLocation();
         CheckpointBarrier barrier = new CheckpointBarrier(id, timestamp, options);
 
-        try {
-            barrier.write(new DataOutputSerializer(1024));
-            fail("should throw an exception");
```
Comment (RocMarshal) [inline flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/CheckpointBarrierTest.java:46]: Should we keep the hint message  by 
```
 assertThatThrownBy(() -> barrier.write(new DataOutputSerializer(1024))).as("should throw an exception")
                .isInstanceOf(UnsupportedOperationException.class);
```

?

**Review by RocMarshal — COMMENTED** [review]
Thanks  @Jiabao-Sun for the contribution.
LGTM +1(non-binding)

File: flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/CheckpointBarrierTest.java (hunk 1 above)
Comment (dawidwys) [reply flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/CheckpointBarrierTest.java:46]: I don't think it adds any value. The `fail("should throw an exception")` is there simply to make sure, the exception is thrown

**Review by dawidwys — APPROVED** [review]
lgtm

**Pull request merged**